### PR TITLE
Export Poll Results with CSV from Admin Panel (Secure Downloading Unsure)

### PR DESCRIPTION
**Pull request by Pyae24**
## Export Poll Results with CSV from Admin Panel
-  https://github.com/khinezarthwe/mmdt_web_app/issues/38

I have done exporting with CSV but not sure whether this has **Secure Downloading** or not.

- I added the `csv` module inside the `admin.py` file and wrote a new function named `export_to_csv` to export poll results in CSV format from admin actions. 

### Admin View
- There are two sample questions inside polls: `Fruits` and `Snacks` and also voted as a user for sample data.

First

- There is an action to export poll results as a csv file inside `Action` and the admin can choose a specific poll question to export.

Second

- The results are then downloaded as a CSV file named (poll_results.csv) when clicked on the `Go` button.

Third

- This is a preview of sample data.

Fourth

### CODE REVIEW

File: mmdt/polls/admin.py [hunk 1]
```diff
@@ -7,6 +9,22 @@ class ChoiceInline(admin.TabularInline):
     model = Choice
     extra = 3
 
+def export_to_csv(modeladmin, request, queryset):
```
Comment (khinezar-mercari) [inline mmdt/polls/admin.py:12]: Please write the function doc in new function.

 for example.
```
def export_to_csv(modeladmin, request, queryset):
    """
    function def
    :param modeladmin: 
    :param request: 
    :param queryset: 
    :return: 
    """
```

File: mmdt/polls/admin.py [hunk 2]
```diff
@@ -7,6 +9,22 @@ class ChoiceInline(admin.TabularInline):
...
+    # Write the headers to csv file
+    writer.writerow(['Question', 'Choice', 'Votes'])
+
+    # Write data
+    for question in queryset:
+        for choice in question.choice_set.all():
+            writer.writerow([question.question_text, choice.choice_text, choice.votes])
+    return response
+
+export_to_csv.short_description = "Export Selected Questions to CSV"
```
Comment (khinezar-mercari) [inline mmdt/polls/admin.py:26]: Let's try to be consistent. Python allows both double quotes and single quotes for strings in the codebase, but it is better for us to be strict with one way. It seems like other file names are written in double quotes, so we should decide which one we should choose. Let's confirm with other codebases and synchronize with them.

**Review by khinezar-mercari — COMMENTED** [review]
Please include the testing evidence and testing data in the pull request (PR), so that other developers can review the user interface (UI) and provide comments based on the evidence provided.

your code is look good, other updates are LGTM.

File: mmdt/polls/admin.py (hunk 1 above)
Comment (Pyae24) [reply mmdt/polls/admin.py:12]: I am not clear what this means ama. Am I supposed to write documentation for this function like this?
```
def export_to_csv(modeladmin, request, queryset):
    """
    export_to_csv
    :param modeladmin: The ModelAdmin instance.
    :param request: The current request.
    :param queryset: The queryset containing selected poll results.
    :return: A CSV file response containing the poll results.
    """

Comment (khinezar-mercari) [reply mmdt/polls/admin.py:12]: yes ! This code base is written by students and didn't put any functions docs, So I am thinking we better start to write the function docs this time.
```
def export_to_csv(modeladmin, request, queryset):
    """
    export the polls data to csv file.
    :param modeladmin: The ModelAdmin instance.
    :param request: The current request.
    :param queryset: The queryset containing selected poll results.
    :return: A CSV file response containing the poll results.
    """
```

**Review by khinezarthwe — APPROVED** [review]

**Pull request merged**